crypto/sm3.mbt

```diff
@@ -77,23 +78,23 @@ fn SM3Context::p_1(x : UInt) -> UInt {
   x ^ urotate_left(x, 15) ^ urotate_left(x, 23)
 }
 
-fn pad(self : SM3Context) -> Array[Byte] {
-  self.msg.push(b'\x80')
-  while self.msg.length() % 64 != 56 {
-    self.msg.push(b'\x00')
+fn pad(self : SM3Context) -> Iter[Byte] {
+  self.msg = self.msg.append(b'\x80')
+  while self.msg.count() % 64 != 56 {
```

Comment:
This is inefficient: the iter will execute everytime when you count

Reply:
changed to this
```moonbit
  let mut cnt = self.msg.count()
  while cnt % 64 != 56 {
    self.msg = self.msg.append(b'\x00')
    cnt += 1
  }
```